### ISSUE TITLE
Issue credential protocol

### ISSUE THREAD

**Issue by TimoGlastra** [hyperledger/aries-framework-javascript#64]
Issue Credential protocol is currently at v1.1. This adds some fields in comparison to v1.0, but is completely backwards compatible with the version in AIP 1.0. 

So we should implement the current version (1.1) from the master branch.

Checklist
- [x] Credential offer exchange
  - Issuer sends credential offer which holder accepts. 
  - A credential record is created and persisted on both sides in a given state: offer-sent, offer-received.
  - Emit events about the state change.
- [x] Credential exchange
  - Issuer sends credential which holder accepts.
  - The credential state is updated accordingly: request-sent, request-received, credential-issued, credential-received.
  - Emit events about the state change.
- [x] Others 
  - Acks: We have implemented just the "Happy Path" part of the [Aries RFC 0317: Please ACK Decorator](https://github.com/hyperledger/aries-rfcs/blob/master/features/0317-please-ack/README.md), but I think it's good enough for now. The spec is still in a PROPOSED state anyway.
  - Threads: We have implemented basic threading mechanism. Other parts of the protocol will be addressed in #128.
  - ~~Encode and decode attachments with a decorator.~~ This doesn't seem as necessary to me right now.
- [x] ~~Alternative beginnings~~ 
  - ~~Start with Credential proposal~~. I've created a new task #129 for this.

Resources
- https://github.com/hyperledger/aries-rfcs/blob/master/features/0036-issue-credential/README.md

Branch with work in-progress
- https://github.com/jakubkoci/aries-framework-javascript/tree/feature/accept-credential-offer

**Comment by jakubkoci** [comment]
This issue depends on #62 which is partially implemented but waiting for some infra preparation to run tests against Indy pool. However, I would like to move forward with this so I start the work and I'll try to fix #62 in parallel.

**Comment by jakubkoci** [comment]
I've started with implementation. Now, I have a better understanding of the task so I've updated the issue with more specific tasks. I would like to create standalone PR for every task to keep our PRs small. I also added a branch where you can see the work in-progress.

I should have the first task pretty soon, perhaps tomorrow.

**Comment by biligunb** [comment]
@jakubkoci @TimoGlastra Good luck guys.
We are using Aries Framework JS for our project currently. And eagerly waiting for the VC and Present Proof protocols.
If I may, can I ask what is the timeline for these 2 to be merged into the master branch?
  - We have previously used ACA-PY and changed to JS. I am sure you guys know VC issue, exchange etc.. is pretty big thing. (thus why i am asking :))

**Comment by TimoGlastra** [comment]
Thank you @biligunb. The first draft PR (#92) for a part of the issue credential protocol is created. Full issue credential protocol should be finished relatively soon.

Present proof will probably take some time. But we'll need it in the coming ~2 months for one of our projects so definitely hope it will be finished by then.

It's nice to hear you're already using the framework for your project. As the framework is very new we don't have a lot of information on using the framework from a consumer point of view. We'll be grateful if you could share some feedback on your experience with using the framework.

**Comment by biligunb** [comment]
@TimoGlastra 
Nice to hear that. 👍

> Present proof will probably take some time

Present Proof in 2 months is not bad too. I understand that not just code but also testing must be done too.

>  share some feedback 

For us we are using JS framework with Chrome browser extension. We could not use `aca-py` in browser so we decided to use JS.
It would be nice to have features like auto-accept invitation, credential offer etc... in JS too 😃

**Comment by jakubkoci** [comment]
Solved by #92 #104 #122
